lib/settings-observer.js

```diff
@@ -45,53 +46,23 @@ class SettingsObserver {
      * @private
      */
     this._currentValue = currentValue;
-
-    /**
-     * Called when the settings file is changed.
-     *
-     * @type {Object}
-     * @private
-     */
-    this._handleChange = this._onChange.bind(this);
-
-    this._init();
-  }
-
-  /**
-   * Initializes this instance.
-   *
-   * @private
-   */
-  _init() {
-    this._settings.on('change', this._handleChange);
   }
-
-  /**
-   * Called when the settings file is changed.
-   *
-   * @private
-   */
-  _onChange() {
-    const oldValue = this._currentValue;
-    const newValue = this._settings.get(this._keyPath);
-
-    try {
-      assert.deepEqual(newValue, oldValue);
-    } catch (err) {
-      this._currentValue = newValue;
-
-      // Call the watch handler and pass in the new and old values.
-      this._handler.call(this, newValue, oldValue);
+  populate() {
+    let newVal = _.get(this._settings._cacheObj, this._keyPath)          
+    if (!_.isEqual(newVal, this._currentValue)) {             
+      this._handler.call(this, newVal, this._currentValue)      
+      this._currentValue = newVal
     }
   }
-
   /**
    * Disposes of this key path observer.
    *
    * @public
    */
   dispose() {
-    this._settings.removeListener('change', this._handleChange);
+    this._settings._watchHandlers = _.remove(this._settings._watchHandlers, (n) => {
```

Comment:
Ugh, we should just delete this._settings._watchHandlers. Calling _.remove doesn't give us the expected result